modules/core/src/main/java/org/apache/synapse/deployers/LocalEntryDeployer.java

```diff
@@ -102,6 +106,50 @@ public String deploySynapseArtifact(OMElement artifactConfig, String fileName,
         return null;
     }
 
+    private void deployEndpointsForHTTPConnection(OMElement artifactConfig, String fileName, Properties properties) {
+
+        OMElement httpInitElement =
+                artifactConfig.getFirstChildWithName(
+                        new QName(SynapseConstants.SYNAPSE_NAMESPACE, HTTP_CONNECTION_IDENTIFIER));
+        if (httpInitElement != null) {
+            OMElement generatedEndpointElement = HTTPConnectionUtils.generateHTTPEndpointOMElement(httpInitElement);
+            deployHTTPEndpointForElement(generatedEndpointElement, fileName, properties);
+        }
+    }
+
+    private void deployHTTPEndpointForElement(OMElement documentElement, String fileName, Properties properties) {
+
+        try {
+            Endpoint ep = EndpointFactory.getEndpointFromElement(documentElement, false, properties);
+
+            //Set the car name
+            ep.setArtifactContainerName(customLogContent);
+            if (ep != null) {
```

Comment:
Can this `ep` become `null`? 
Doesn't the 
`Endpoint ep = EndpointFactory.getEndpointFromElement(documentElement, false, properties);`
always return an endpoint?